### PR TITLE
contributing: front new nomenclature

### PR DESCRIPTION
**Pull request by nicolaswurtz**

### CODE REVIEW

**Review by axrolld — APPROVED** [review]
🚀

File: content/docs/guides/contribute/conventions/_index.fr.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+## Règles générales
+
+- Expliquez ce que vous faites et pourquoi.
+- Documentez le nouveau code.
+- Ajoutez des tests clairs et simples.
+- Décomposez le travail en morceaux intelligibles.
+- Prenez le temps de choisir de bons noms.
+  Évitez les abréviations peu connues.
+- **Contrôle et cohérence de la réutilisation du code de tiers** : Ajoutez une dépendance que si elle est absolument nécessaire.
```
Comment (Math-R) [inline content/docs/guides/contribute/conventions/_index.fr.md:18]: que si => seulement si

File: content/docs/guides/contribute/conventions/front.en.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+> **Le SCSS n'est pas scopé**
+>
+> Un fichier `.scss` enfoui dans l'arborescence ne vous garantit pas que les classes contenues soient seulement accessibles à cet endroit, y compris par import react (formellement interdit au passage : vous devez utiliser l'import SCSS), toutes les classes déclarées sont accessibles partout.
+>
+> Préférez un choix judicieux de nom de classe racine pour un module donné et utilisez l'arborescence possible dans le fichier SCSS.
+
+> **Les liens des imports doivent être absolus**
+>
+> Vous devez utiliser le <u>chemin complet</u> pour tous vos imports, sans exception, même si le fichier à importer se trouve dans le même répertoire.
```
Comment (Math-R) [inline content/docs/guides/contribute/conventions/front.en.md:70]: To modify : modules are de facto libraries of components, so import between component inside the same module should be relative

File: content/docs/guides/contribute/conventions/_index.en.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+- Explain what you're doing and why.
+- Document new code with doc comments.
+- Include clear, simple tests.
+- Break work into digestible chunks.
+- Take the time to pick good names.
+  Avoid non well-known abbreviations.
+- **Control and consistency over 3rd party code reuse**: Only add a dependency if it is absolutely necessary.
+  Every dependency we add decreases our autonomy and consistency.
+- **Don't re-invent every wheel**: As a counter to the previous point, don't re-invent everything at all costs.
```
Comment (axrolld) [inline content/docs/guides/contribute/conventions/_index.en.md:20]: Based on [this transaltion](https://www.linguee.fr/francais-anglais/traduction/r%C3%A9inventer+la+roue.html)
```suggestion
- **Don't reinvent the wheel**: As a counter to the previous point, don't reinvent everything at all costs.
```

File: content/docs/guides/contribute/conventions/_index.fr.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+- Ajoutez des tests clairs et simples.
+- Décomposez le travail en morceaux intelligibles.
+- Prenez le temps de choisir de bons noms.
+  Évitez les abréviations peu connues.
+- **Contrôle et cohérence de la réutilisation du code de tiers** : une dépendance est ajoutée seulement si elle est absolument nécessaire.
+  Chaque dépendance ajoutée diminue notre autonomie et notre cohérence.
+- **Ne pas réinventer la roue** : En opposition au point précédent, ne réinventez pas tout à tout prix.
+  S'il existe une dépendance dans l'écosystème qui est le standard "de-facto", nous devrions fortement envisager de l'utiliser.
+- Plus de code et de recommandations générales dans le dépôt principal [CONTRIBUTING.md](https://github.com/osrd-project/osrd).
+- Demandez toute l'aide dont vous avez besoin !
```
Comment (axrolld) [inline content/docs/guides/contribute/conventions/_index.fr.md:23]: Missing dashes
```suggestion
- Évitez les abréviations peu connues.
- **Contrôle et cohérence de la réutilisation du code de tiers** : une dépendance est ajoutée seulement si elle est absolument nécessaire.
- Chaque dépendance ajoutée diminue notre autonomie et notre cohérence.
- **Ne pas réinventer la roue** : En opposition au point précédent, ne réinventez pas tout à tout prix.
- S'il existe une dépendance dans l'écosystème qui est le standard "de-facto", nous devrions fortement envisager de l'utiliser.
- Plus de code et de recommandations générales dans le dépôt principal [CONTRIBUTING.md](https://github.com/osrd-project/osrd).
- Demandez toute l'aide dont vous avez besoin !
```

File: content/docs/guides/contribute/conventions/back.en.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+# Python
+
+Python code is used for some packages and integration testing.
+
+- Follow [the Zen of Python](https://www.python.org/dev/peps/pep-0020/).
+- Code is linted with [flake8](https://github.com/csachs/pyproject-flake8).
+- Code is formatted with [Black](https://github.com/psf/black).
+- Imports are sorted with [Isort](https://github.com/PyCQA/isort).
+- Python tests are written using [pytest](https://docs.pytest.org/).
```
Comment (axrolld) [inline content/docs/guides/contribute/conventions/back.en.md:15]: Didn't we get rid of python now that api is dead ?

File: content/docs/guides/contribute/conventions/_index.fr.md (hunk 4 above)
Comment (axrolld) [reply content/docs/guides/contribute/conventions/_index.fr.md:23]: Or maybe they're not supposed to be here, in that case there are too many dashes in the english version

File: content/docs/guides/contribute/conventions/front.en.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
+---
+title: "Front-end"
+linkTitle: "Front-end"
+description: "Coding style guide and best practices for front-end"
```
Comment (axrolld) [inline content/docs/guides/contribute/conventions/front.en.md:4]: This is supposed to be the english page but the entire text is in french, maybe a bad copy-paste ?

**Review by axrolld — CHANGES_REQUESTED** [review]

**Review by Math-R — APPROVED** [review]

File: content/docs/guides/contribute/conventions/front.en.md (hunk 6 above)
Comment (nicolaswurtz) [reply content/docs/guides/contribute/conventions/front.en.md:4]: Nope, as this section is currently on rewrite/conception (along actual front refacto), we write it in french, and translation will come after stabilization !

Comment (nicolaswurtz) [reply content/docs/guides/contribute/conventions/front.en.md:4]: Resolve is at your discretion :heart:

File: content/docs/guides/contribute/conventions/back.en.md (hunk 5 above)
Comment (nicolaswurtz) [reply content/docs/guides/contribute/conventions/back.en.md:15]: I've not updated this text, you should ask dev mainteners :slightly_smiling_face:

Comment (nicolaswurtz) [reply content/docs/guides/contribute/conventions/back.en.md:15]: Resolve is at your discretion

Comment (axrolld) [reply content/docs/guides/contribute/conventions/back.en.md:15]: I asked @flomonster, there is still some python so we should keep this

File: content/docs/guides/contribute/conventions/front.en.md (hunk 6 above)
Comment (axrolld) [reply content/docs/guides/contribute/conventions/front.en.md:4]: Hmm seems weird to me but ok 🤔

**Pull request merged**